APISpec.md

```diff
@@ -0,0 +1,144 @@
+# Todo API Spec
+
+## API Resources
+
+* **Todo** - A Todo represents a task to be completed.
+
+## Data Tables
+
+### Todo:
+```
+ID - Guid (PK)
+Title - String
+Status - Enumified String
+CreationDate - Datetime
+DueDate - Datetime
+```
+
+## Data Models
+
+### Todo
+```
+ID - Guid
+Title - String
+Status - Status
+CreationDate - Datetime
+DueDate - Datetime
+```
+
+### Status
+```
+OPEN
+IN_PROGRESS
+COMPLETE
+```
+
+## Endpoints
+
+### Create New Todo
+---
+
+```
+POST /todo
+```
+
+Body: 
+```
+{
+    "meta": {},
+    "data": {
+        "title": "Buy groceries",
+        "dueDate": "2022-04-01T13:02:37.000Z"
+    }
+}
+```
+
+**Response** - `204 No Content`
```

Comment:
Often times when creating a resource it's useful for the client to receive a copy of the resource created. Additionally idiomatic HTTP indicates that the appropriate response to creating a resource is `201 Created`